tests/boards/nxp_s32/swt/src/main.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2022 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/drivers/watchdog.h>
+
+#define DEVICE_FOR_DT_COMPAT(n, compat) DEVICE_DT_GET(DT_INST(n, compat)),
```

Comment:
is there anything about this test that is board specific? Or could a new test be added to `tests/drivers/watchdog`, that verifies the watchdog callback feature without resetting the SOC?

Reply:
+1 with the idea of extending existing tests / creating new generic test, for this use-case

Reply:
This test isn't exactly board specific. Yes, it verifies the watchdog callback feature without resetting the SOC.
I can put it in `tests/drivers/watchdog` but I'm not sure whether vendors need this feature or not. So I put it in `tests/boards`. Also, this test isn't necessary when `s32z270dc2_r52 ` board supports running from flash.

Reply:
I think unless the test needs to be specific to a board, it should be placed in `tests/drivers/watchdog`. This will at least make it possible for other vendors to use the test if desired.